### PR TITLE
Refactor registry etcd to storage

### LINKED ISSUE #17546: Registry Renaming etcd -> storage
Now that all elements in the registry have been switched to use the storage layer we should probably move to rename the /etcd\* to something more generic, as we work to shard to different backends.  

I still have to finish up some other transformations 1st, but this is a placeholder. 

My vote might be keyval.  

/cc @jayunit100 @lavalamp @wojtek-t @rrati

### PR DESCRIPTION
**Pull request by timothysc**
Fixes #17546 

Simple shuffle on naming so any sane new person entering the code base can understand what the actual etcd dependencies are.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37039)

### CODE REVIEW

**Comment by timothysc** [comment]
@lavalamp @wojtek-t before I go through rebase purgatory, any comments?

**Comment by wojtek-t** [comment]
I think I'm fine with this PR, but @lavalamp had some strong opinion about it in the past. 
@lavalamp - any thoughts on it?

**Comment by timothysc** [comment]
> but @lavalamp had some strong opinion about it in the past. 

really?  I believe he approved the initial PR's of this change a while ago.

**Comment by wojtek-t** [comment]
If he's fine with that, then ok (I'm definitely not good at naming).

**Comment by timothysc** [comment]
@lavalamp If you have a moment to quick review, this is simply tech debt rename shuffling.

**Comment by timothysc** [comment]
@wojtek-t this has been open for 25 days, can we move forwards using the #24061 as the cited precedent here?

**Comment by wojtek-t** [comment]
@timothysc - yes, please rebase and I will take one more pass then

**Comment by timothysc** [comment]
@wojtek-t pushed a rebase, hopefully no issues.  I appears the api jiggery always gets hit every time now.

**Comment by timothysc** [comment]
@k8s-bot test this #IGNORE

File: federation/cmd/federation-apiserver/app/core.go [hunk 1]
```diff
@@ -33,20 +33,21 @@ import (
 	"k8s.io/kubernetes/pkg/api/rest"
 	"k8s.io/kubernetes/pkg/apimachinery/registered"
 	"k8s.io/kubernetes/pkg/genericapiserver"
-	configmapetcd "k8s.io/kubernetes/pkg/registry/core/configmap/etcd"
-	eventetcd "k8s.io/kubernetes/pkg/registry/core/event/etcd"
-	namespaceetcd "k8s.io/kubernetes/pkg/registry/core/namespace/etcd"
-	secretetcd "k8s.io/kubernetes/pkg/registry/core/secret/etcd"
-	serviceetcd "k8s.io/kubernetes/pkg/registry/core/service/etcd"
+	configmapstore "k8s.io/kubernetes/pkg/registry/core/configmap/storage"
```
Comment (wojtek-t) [inline federation/cmd/federation-apiserver/app/core.go]: We generally have a convension that the alias is concatentation of last few parts of the path. So it should be "configmapstorage".

Same in other places.

Comment (wojtek-t) [inline federation/cmd/federation-apiserver/app/core.go]: Or alternatively, we can change the directory to "store" instead, but I think that option is worse.

File: pkg/registry/OWNERS [hunk 2]
```diff
@@ -2,3 +2,4 @@ assignees:
   - lavalamp
   - smarterclayton
   - wojtek-t
+  - timothysc
```
Comment (wojtek-t) [inline pkg/registry/OWNERS]: Please send a separate PR for it - we should hide those in such large PRs.

File: pkg/registry/core/service/allocator/storage/storage.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ var (
 	errorUnableToAllocate = errors.New("unable to allocate")
 )
 
-// Etcd exposes a service.Allocator that is backed by etcd.
+// Etcd exposes a service.Allocator
```
Comment (wojtek-t) [inline pkg/registry/core/service/allocator/storage/storage.go]: Should we also rename Etcd to Storage or sth like that?

**Review by wojtek-t — COMMENTED** [review]
Also - please squash commits.

File: pkg/registry/OWNERS (hunk 2 above)
Comment (timothysc) [inline pkg/registry/OWNERS]: Sure I'll remove.

File: pkg/registry/core/service/allocator/storage/storage.go (hunk 3 above)
Comment (timothysc) [inline pkg/registry/core/service/allocator/storage/storage.go]: Agreed, but would be ok to knock on PR that one?

File: federation/cmd/federation-apiserver/app/core.go (hunk 1 above)
Comment (wojtek-t) [inline federation/cmd/federation-apiserver/app/core.go]: Can you please open an issue to clear this up then?

**Comment by wojtek-t** [comment]
Please squash commits and I will lgtm.

File: pkg/registry/OWNERS (hunk 2 above)
Comment (timothysc) [inline pkg/registry/OWNERS]: Done.

File: federation/cmd/federation-apiserver/app/core.go (hunk 1 above)
Comment (timothysc) [inline federation/cmd/federation-apiserver/app/core.go]: Done. https://github.com/kubernetes/kubernetes/issues/38874

File: pkg/registry/core/service/allocator/storage/storage.go (hunk 3 above)
Comment (timothysc) [inline pkg/registry/core/service/allocator/storage/storage.go]: Done. https://github.com/kubernetes/kubernetes/issues/38874

**Comment by timothysc** [comment]
> Please squash commits and I will lgtm.

@wojtek-t done.

**Comment by lavalamp** [comment]
Sorry about radio silence--I was OOO most of last month. I approve of this name change, thank you very much!

**Comment by timothysc** [comment]
@k8s-bot bazel test this

**Comment by timothysc** [comment]
@ixdy @spxtr @zmerlynn  something about this bazel build is messed up, all local build + verification, etc. work fine.

**Comment by spxtr** [comment]
@mikedanese

**Comment by ixdy** [comment]
https://github.com/kubernetes/kubernetes/pull/39483 merged 30m ago and removed pkg/auth/authenticator/bearertoken, among other things.

you'll probably need to merge/rebase and re-gazel.

**Comment by mikedanese** [comment]
Likely a silent merge conflict.

**Comment by timothysc** [comment]
self applying per @wojtek-t comment b/c he's out and I've rebased this thing too many times.

**Comment by timothysc** [comment]
@ixdy did all that and now a new issue.

**Comment by timothysc** [comment]
@k8s-bot bazel test this

File: test/integration/auth/rbac_test.go [hunk 4]
```diff
@@ -1,5 +1,3 @@
-// +build integration,!no-etcd
```
Comment (mikedanese) [inline test/integration/auth/rbac_test.go]: Why is this removed? Should it still have an integration tag?

**Comment by mikedanese** [comment]
We aren't compiling with the integration tag. After removing the build tag, k8s.io/kubernetes/test/integration/auth only compiles with the integration build tag set. You can confirm this by running `go test ./test/integration/auth`. If you readd the build tag, you will need to manually remove the ./test/integration/auth/BUILD file.

**Comment by mikedanese** [comment]
@timothysc looking into most recent failure, just a sec... :(

**Comment by mikedanese** [comment]
you needed to manually remove the ./test/integration/auth/BUILD file.

**Comment by mikedanese** [comment]
@k8s-bot unit test this
